### PR TITLE
Add retry logic for docker commands

### PR DESCRIPTION
**Pull request by flooose**
## Summary

Because our github actions sometimes run into rate limiting from docker hub and AWS ECR, we want  to retry when this happens.

## Related Issues

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/89959

### CODE REVIEW

**Review by RachalCassity — DISMISSED** [review]

File: docker-compose.test.yml [hunk 1]
```diff
@@ -11,7 +11,6 @@ services:
   web:
     build:
       context: .
-      target: development
```
Comment (flooose) [inline docker-compose.test.yml:14]: See the [commit message](https://github.com/department-of-veterans-affairs/vets-api/pull/17877/commits/b4cc2248ef5979faf0a13d0b6ecac3b77c276b95) about this one.

**Comment by rmtolmach** [comment]
Sometimes this fails on the **Audit Service Tags** check as well. Does it make sense to add a retry to that check as well?

**Comment by flooose** [comment]
> Sometimes I see the`toomanyrequests` error on the **Audit Service Tags** check as well. Does it make sense to add a retry to that check as well?

Yes! Done, and thank you :)

File: .github/workflows/audit_service_tags.yml [hunk 2]
```diff
@@ -45,20 +45,19 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v3
 
-      - name: Build Docker Image
-        uses: docker/build-push-action@v6
+      - name: Setup Caching for Docker Compose
+        uses: yu-ichiro/spin-up-docker-compose-action@main
```
Comment (rmtolmach) [inline .github/workflows/audit_service_tags.yml:49]: Did Eric recommend changing this to not be `main`?

Comment (flooose) [reply .github/workflows/audit_service_tags.yml]: Yes! Although we did decide to go another way with this. The commit hashes should be there for the other gh-action that we're using instead.

File: .github/workflows/audit_service_tags.yml [hunk 3]
```diff
@@ -61,8 +61,13 @@ jobs:
...
 
       - name: Setup Database
-        run: |
-             docker compose -f docker-compose.test.yml run web bash \
+        uses: nick-fields/retry@7152eba30c6575329ac0576536151aca5a72780e # v3.0.0
+        with:
+          timeout_minutes: 20
+          retry_wait_seconds: 3 # Seconds
+          max_attempts: 3
+          command: |
```
Comment (rmtolmach) [inline .github/workflows/audit_service_tags.yml:69]: Do we also need to use the `retry_on` input to only retry after an error? I assumed that would be the default... but looking at [the docs](https://github.com/nick-fields/retry?tab=readme-ov-file#only-retry-after-error) maybe we need to specify it?

Comment (flooose) [reply .github/workflows/audit_service_tags.yml:69]: It says it's optional and the default is to retry on error *and* timeout, which _seems_ sane to me. What are your thoughts though?

**Review by rmtolmach — APPROVED** [review]

File: .github/workflows/audit_service_tags.yml (hunk 3 above)
Comment (rmtolmach) [reply .github/workflows/audit_service_tags.yml:69]: oh yeah.. I see that now :)

**Pull request merged**